src/components/form/SignUp.tsx

```diff
@@ -0,0 +1,152 @@
+import { FC, useState } from "react";
+import axios from "axios";
+
+import { useForm, SubmitHandler } from "react-hook-form";
+interface SignUpFormData {
+    name: string;
+    email: string;
+    password: string;
+    confirmPassword: string;
+}
+
+const SignUp: FC = () => {
+
+    const [isLoading, setIsLoading] = useState(false);
+    const {
+        register,
+        handleSubmit,
+        formState: { errors, isDirty, isValid },
+        watch,
+    } = useForm<SignUpFormData>({
+        mode: "onChange",
+        defaultValues: {
```

Comment:
Move default values to separate file like (defaultValues)
then 
export const signupDefaultValues = {...}
then use these values in your file